src/plugins/discover/public/application/angular/discover.js

```diff
@@ -896,6 +921,19 @@ function discoverController($element, $route, $scope, $timeout, $window, Promise
       });
     });
 
+    if ($scope.useNewFieldsApi) {
+      // get some columns to display
+      const allColumns = getIndexPatternFieldList($scope.indexPattern, counts).map(
```

Comment:
If we're not displaying `_source` anymore, what columns should we show? This takes the first few fields from the index pattern and creates columns (this should probably be read from the fields response).

Reply:
Maybe wie could rename `_source` to `fields` in the table header, and display a number of fields in the way `_source is` displayed today? this would be the similar approach like it is today, only a different data source ... but yes, also preselecting columns as default would be an option, however in that case, the user might need more action to get to the columns he wants to display (removing the default columns)